core/src/test/scala/com/salesforce/op/ModelInsightsTest.scala

```diff
@@ -651,4 +722,40 @@ class ModelInsightsTest extends FlatSpec with PassengerSparkFixtureTest with Dou
       f.cramersV.isEmpty shouldBe true
     }
   }
+
+  it should "correctly return the descaled coefficient for linear regression, " +
+    "when standardization is on" in {
+
+    // Since 5000 & 1 are always returned as the coefficients of the model
+    // trained on unstandardized data and we can analytically calculate
+    // the scaled version of them by the linear regression formula, the coefficients
+    // of the model trained on standardized data should be within a small distance of the analytical formula.
+
+    // difference between the real coefficient and the analytical formula
+    //     return Array(descaledsmallCoeff, originalsmallCoeff, descaledbigCoeff, orginalbigCoeff)
+    val coeffs = getFeatureImp(standardizedLinpred, unstandardizedLinpred, linRegDF._3)
+    val descaledsmallCoeff = coeffs(0)
+    val originalsmallCoeff = coeffs(1)
+    val descaledbigCoeff = coeffs(2)
+    val orginalbigCoeff = coeffs(3)
+    val absError = math.abs(orginalbigCoeff * math.sqrt(smallFeatureVariance) / labelStd - descaledbigCoeff)
+    val absError2 = math.abs(originalsmallCoeff * math.sqrt(bigFeatureVariance) / labelStd - descaledsmallCoeff)
+    absError < 0.01 shouldBe true
```

Comment:
Can you make these tolerances a single value for the whole test class?

Also, if you're comparing a floating point calculation like this, it's usually easier to do a relative comparison. You can often use smaller tolerances if you do that too.
For example, if you have a calculation result `res` that you think should be equal to a ground truth value `truth`, then you can compare something like `abs(res - truth) * 2 / (res + truth) < tolerance` which is just the difference over the average value. If you compare the differences directly, then you often have to use different tolerances for different comparisons (like you do in the next test).

Reply:
Oh, I meant making a single variable like `val tol = 0.05` so that we can adjust them all at once if this ends up being flaky in the future.

Reply:
Also, can you check what the average difference is? I would guess you can go much lower than your absolute difference tolerance now that you've converted them to relative differences.